File: ClientesOrionTekBackEnd/Models/Address.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ClientesOrionTekBackEnd.Models;

public class Address
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long AddressId { get; set; }

    [Required]
    public long ClientId { get; set; }

    [Required]
    [MaxLength(255)]
    public string Street { get; set; } = string.Empty;

    [Required]
    [MaxLength(100)]
    public string City { get; set; } = string.Empty;

    [Required]
    [MaxLength(100)]
    public string Sector { get; set; } = string.Empty;

    [Required]
    [MaxLength(20)]
    public string HouseNumber { get; set; } = string.Empty;

    [ForeignKey(nameof(ClientId))]
    public virtual Client? Client { get; set; }
}

File: ClientesOrionTek/app/src/main/java/com/example/clientesoriontek/data/local/entity/Address.kt
package com.example.clientesoriontek.data.local.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "addresses",
    foreignKeys = [
        ForeignKey(
            entity = Client::class,
            parentColumns = ["id"],
            childColumns = ["clientId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["clientId"])]
)
data class Address(
    @PrimaryKey(autoGenerate = true) val addressId: Long = 0,
    val clientId: Long,
    val street: String,
    val city: String,
    val sector: String,
    val houseNumber: String
)
